tutorials/iot/graphical-snaps/graphical-snaps.md

```diff
@@ -0,0 +1,608 @@
+...
+```
+
+Add a `plugs` stanza:
+```yaml
+...
+plugs:
+   wayland-socket-dir:
+    content: wayland-socket-dir
+    interface: content
+    target: $SNAP_DATA/wayland
+    default-provider: mir-kiosk
+...
+```
+   
+This additional snippet causes snapd to bind-mount the mir-kiosk Wayland socket directory into the glmark2-wayland’s namespace, in a location of its desire: $SNAP_DATA/wayland. Then we update the XDG_RUNTIME_DIR variable to match.
+
+### The full YAML file:
+
+```yaml
+name: iot-example-graphical-snap
+version: 0.1
+summary: GLMark2 on Wayland
+description: |
+  GLMark2 on Wayland
+confinement: strict
+grade: devel
+
+apps:
+  glmark2-wayland:
+    command: "bash -c 'XDG_RUNTIME_DIR=$SNAP_DATA/wayland LIBGL_DRIVERS_PATH=$(find $SNAP/usr/lib/ -name dri) $SNAP/usr/bin/glmark2-wayland'"
+    plugs:
+      - opengl
+      - wayland
+
+parts:
+  glmark2-wayland:
+    plugin: nil
+    stage-packages:
+      - glmark2-wayland
+
+passthrough:
+  layout:
+    /usr/share/glmark2:
+      bind: $SNAP/usr/share/glmark2
+
+plugs:
+   wayland-socket-dir:
+    content: wayland-socket-dir
+    interface: content
+    target: $SNAP_DATA/wayland
+    default-provider: mir-kiosk
+```
+
+Check this builds locally:
+```bash
+snapcraft cleanbuild
+```
+## Building for different architectures
+duration: 10:00
+
+One day, perhaps, snapcraft will fully support cross building with the `--target-arch` option. But getting that to work is beyond the scope of this tutorial. Instead we'll make use of Launchpad's builders to build the snap for all architectures (including the one your device provides).
+
+Create a github repository for your snap and push your changes to the snap project there:
+
+```bash
+git commit -a
+git remote remove origin
+git remote add origin https://github.com/<project>/<repo>.git
+git push -u origin master
+```
+
+Now [setup your build on Launchpad](https://docs.snapcraft.io/build-snaps/ci-integration). *Note that you will need to use the same snap name in the store as in name in `name:`, so chose something unique to make your life easier.*
+
+Don't bother with publishing to the store (yet) you can download the snap and deploy it as follows:
+
+On your desktop go to the snap webpage (e.g. https://code.launchpad.net/~alan-griffiths/+snap/iot-example-graphical-snap), find the build for your device architecture and download it and copy to your device. For example:
+```bash
+wget https://code.launchpad.net/~alan-griffiths/+snap/iot-example-graphical-snap/+build/226518/+files/iot-example-graphical-snap_0.1_arm64.snap
+scp iot-example-graphical-snap_0.1_arm64.snap   alan-griffiths@192.168.1.159:~
+```
+On your ssh session to your device:
+```bash
+snap install --dangerous ./iot-example-graphical-snap_0.1_arm64.snap 
+snap connect iot-example-graphical-snap:wayland-socket-dir  mir-kiosk:wayland-socket-dir
```

Comment:
`sudo` is likely required for the two above.

Reply:
not on Core actually

Reply:
Oh really? `console-setup` must be doing something to allow the user it creates to use snap without logging in, then.